agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -538,26 +540,20 @@ func (dg *dockerGoClient) CreateContainer(ctx context.Context,
 }
 
 func (dg *dockerGoClient) createContainer(ctx context.Context,
-	config *docker.Config,
-	hostConfig *docker.HostConfig,
+	config *dockercontainer.Config,
+	hostConfig *dockercontainer.HostConfig,
 	name string) DockerContainerMetadata {
-	client, err := dg.dockerClient()
+	client, err := dg.sdkDockerClient()
 	if err != nil {
 		return DockerContainerMetadata{Error: CannotGetDockerClientError{version: dg.version, err: err}}
 	}
 
-	containerOptions := docker.CreateContainerOptions{
-		Config:     config,
-		HostConfig: hostConfig,
-		Name:       name,
-		Context:    ctx,
-	}
-	dockerContainer, err := client.CreateContainer(containerOptions)
+	dockerContainer, err := client.ContainerCreate(ctx, config, hostConfig, &network.NetworkingConfig{}, name)
 	if err != nil {
 		return DockerContainerMetadata{Error: CannotCreateContainerError{err}}
 	}
 
-	return MetadataFromContainer(dockerContainer)
+	return dg.containerMetadata(ctx, dockerContainer.ID)
```

Comment:
This should be `MetadataFromContainer`. It was changed in this PR-- https://github.com/aws/amazon-ecs-agent/pull/1500

Reply:
I think we may need to inspectContainer before we can get metadata for the SDK functions because the ContainerCreate call returns just the container ID, whereas for the go-dockerclient, ContainerCreate returned a *container